stdlib/public/core/ClosedRange.swift

```diff
@@ -488,3 +488,20 @@ extension ClosedRange {
 // shorthand. TODO: Add documentation
 public typealias CountableClosedRange<Bound: Strideable> = ClosedRange<Bound>
   where Bound.Stride : SignedInteger
+
+extension ClosedRange: Decodable where Bound: Decodable {
+  public init(from decoder: Decoder) throws {
+    var container = try decoder.unkeyedContainer()
+    let lowerBound = try container.decode(Bound.self)
+    let upperBound = try container.decode(Bound.self)
+    self = lowerBound...upperBound
```

Comment:
Using the `...` operator raises an interesting question. Should invalid range deserialization result in a trap or `throw`? If you obtain a JSON with a pair where lower bound is greater than upper bound, and try to deserialize it, implementation as proposed would simply trap and the invoking code has no way of handling this, as in, how would you make sure that the serialized range is correct before trying to deserialize it?
I think, the range invariant should be checked here and the result constructed using `init(uncheckedBounds:)`. Invalid range scenario should result in an error being `throw`n, which creates another problem: introduction of an error type to the standard library. 🤔
/cc @airspeedswift

Reply:
Ahh.. @airspeedswift pointed out that we already have just the type: `DecodingError.dataCorrupted` might be the right choice of an error value to throw.

Reply:
Made the required changes 👍🏻